providers/aws/resources/aws_s3.go

```diff
@@ -198,7 +204,7 @@ func (a *mqlAwsS3Bucket) policy() (*mqlAwsS3BucketPolicy, error) {
 	}
 
 	// no bucket policy found, return nil for the policy
-	return &mqlAwsS3BucketPolicy{}, nil
+	return nil, errors.New("no policy found")
```

Comment:
why are we returning an error and not just nil?

Reply:
hmm, good point. ill check that

Reply:
I think a NilData would be great because no policy is defined.

Reply:
fyi: we might have to set the field in this case (ie: `TValue[T]{State: StateIsNull | StateIsSet, Error: err}`). The hard thing here is that I had a hard time detecting with go if something is actually nil, because the nil gets wrapped in other types. I'll help solve this in the future to make it clear, but for now just set the TValue manually and return nil, nil if you need this :)

Reply:
👍  pushed up the fix

Reply:
ugh, this resource is giving me issues when nil
```
cnquery> aws.s3.buckets { policy.id}
1 error occurred:
	* cannot cast resource to resource type: <nil>
	* 
	````